specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+# Adaptive Profile
+
+This profile defines a subset of the QIR specification to support a coherent set
+of functionalities and capabilities that might be offered by a quantum backend.
+Like all profile specifications, this document is primarily intended for
+[compiler backend](https://en.wikipedia.org/wiki/Compiler#Back_end) authors as
+well as contributors to the [targeting
+stage](../Compilation_And_Targeting.md#targeting) of the QIR compiler.
+
+The adaptive profile specifies supersets of the base profile that enable
+control flow based on mid-circuit measurements and classical computations
+within coherence times. A backend can support this profile by supporting
+a minimum set of features beyond the base profile and can opt in for features
+beyond that. This marks the first attempt at defining a profile greater in scope
+than the base profile but still limited when compared to the full QIR specification.
+
+Being a superset of the base profile means that these 3 *capabilities* must be
+implemented (a few restrictions are removed from the base profile):
+
+1. An adaptive profile program can execute a sequence of quantum instructions
+   that transform the quantum state.
+2. An adaptive profile program can measure any qubit used in the program and
+   thus a backend must be able to measure the state of all qubits for use in
+   program output generation.
+3. An adaptive profile program must produce one of the specified [output schemas](../output_schemas/).
```

Comment:
```suggestion
3. The exact sequence of instructions applied when executing an adaptive profile program may depend on measurement outcomes. Specifically, this means that forward branching, meaning a conditional jump to a new block of instructions (expressed as `br` instruction), must be supported. Furthermore, a conversion to turn a measurement result into a boolean value (`__quantum__rt__read_result__body`) must be supported, and it must be possible to perform basic classical logic on boolean values.
4. An adaptive profile program must produce one of the specified [output schemas](../output_schemas/).
```

Reply:
Edit: The basic classical logic that need to be supported on boolean values needs to be listed explicitly somewhere.